package/MDAnalysis/analysis/MSD.py

```diff
@@ -168,24 +161,45 @@ def __init__(self, universe, select_list, start_frame, stop_frame,
             If true, include tau values after stop_frame in calculation (this
             is useful for avoiding data loss when breaking large calculations
             across multiple batch jobs)
+        verbose : bool
+            If true, outputs logger.info, else outputs logger.error
         """
 
         self.universe = universe
         self.select_list = select_list
-        if isinstance(select_list, str):
-            self.select_list = select_list.split(',')
-        print(self.select_list)
-        self.start_frame = int(start_frame)
-        self.stop_frame = int(stop_frame)
-        self.dt_restart = int(dt_restart)
+
+        self.start_frame, self.stop_frame, self.dt_restart = \
+            self.universe.trajectory.check_slice_indices(
+                          start_frame,stop_frame,dt_restart)
+
         self.out_name = out_name
+
+        # initializing logger object
+        logging.basicConfig(level=logging.INFO if verbose
+                            else logging.WARNING)
+        self.logger = logging.getLogger(__name__)
+
+        self.logger.info(self.select_list)
+        
         # handling short simulation cases
         n_frames = int(stop_frame) - int(start_frame) + 1
         if int(len_msd) > int(n_frames):
             len_msd = int(n_frames)
+            self.logger.warning(
+                "Correcting len_msd to the number for frames in the trajectory. ")
```

Comment:
This should raise an exception instead of silently changing the input. Not everyone looks into the logs especially in an interactive session. It might take the user quite some time to figure out that is trajectory isn't long enough, since the code he uses looks correct and no visible warning is printed.